src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const half = 1 / 2;
+    const perimetr = this.a + this.b + this.c;
+    const halfPerimeter = half * perimetr;
```

Comment:
two variables are redundant
```suggestion
    const semiPerimeter = (this.a + this.b + this.c) / 2;
```